server/db/db.js

```diff
@@ -4,68 +4,114 @@
       if (err) {
+        log.warn('error saving user: ' + err);
+        // TODO send the item to a retry queue?
         return cb(err);
       }
-      conn.query(query, [buf(fxaId), email, oauthToken], function(err) {
-        if (err) {
-          log.warn('error saving user: ' + err);
-          // TODO send the item to a retry queue?
-        }
-        conn.release();
-        cb(err);
-      });
+      log.info('created user');
+      cb(err);
     });
   },
   getUserById: function(fxaId, cb) {
-    var query = 'SELECT email, oauth_token FROM users WHERE fxa_id = ?';
-    _getConn(function(err, conn) {
-      // TODO handle conn err?
-      conn.query(query, buf(fxaId), function(err, result) {
-        if (err) {
-          log.warn('error retrieving user: ' + err);
-        }
-        conn.release();
+    _trace('db.getUserById', arguments);
+    var query = 'SELECT email, oauthToken FROM users WHERE fxaId = ?';
+    pool.query(query, fxaId, function(err, result) {
+      if (err) {
+        log.warn('error retrieving user: ' + err);
+      }
 
-        cb(err, result);
-      });
+      cb(err, result);
+    });
+  },
+  getPaginatedVisits: function(fxaId, visitId, count, cb) {
+    _trace('db.getPaginatedVisits', arguments);
+    var query = 'SELECT id, url, urlHash, title, visitedAt ' +
+                'FROM visits WHERE fxaId = ? ' +
+                'and visitedAt > (SELECT visitedAt from visits where id = ?) ' +
+                'ORDER BY visitedAt LIMIT ?';
+    pool.query(query, [fxaId, visitId, count], function(err, results) {
+      cb(err, results);
+    });
+  },
+  getVisits: function(fxaId, count, cb) {
+    _trace('db.getVisits', arguments);
+    var query = 'SELECT id, url, urlHash, title, visitedAt ' +
+                'FROM visits WHERE fxaId=? ORDER BY visitedAt LIMIT ?';
+    pool.query(query, [fxaId, count], function(err, results) {
+      cb(err, results);
+    });
+  },
+  getVisit: function(fxaId, visitId, cb) {
+    _trace('db.getVisit', arguments);
+    // important: MySQL enforces that the user with id `fxaId` is the user with visit `visitId`
+    var query = 'SELECT id, url, urlHash, title, visitedAt ' +
+                'FROM visits WHERE id = ? AND fxaId = ?';
+    pool.query(query, [visitId, fxaId], function(err, r) {
+      if (err) {
+        log.warn('error getting visit: ' + err);
+      }
+      cb(err, r && r[0]);
+    });
+  },
+  // TODO if the url and visitedAt are the same, should we just discard the record?
+  // ...maybe just deal with it later
+  createVisit: function(fxaId, visitId, visitedAt, url, title, cb) {
+    _trace('db.createVisit', arguments);
+    var urlHash = crypto.createHash('sha1').update(url).digest('hex').toString();
+    var query = 'INSERT INTO visits (id, fxaId, rawUrl, url, urlHash, title, visitedAt) ' +
+                'VALUES (?, ?, ?, ?, ?, ?, ?)';
+    pool.query(query, [visitId, fxaId, url, url, urlHash, title, visitedAt], function(err, results) {
```

Comment:
I'll add a routing validation to cap the length at 128 chars.

SEO people suggest keeping the title under 70 or so chars, so I expect most content publishers will put the most important stuff earlier in the title: http://blog.powermapper.com/blog/post/Page-Title-Length-for-Search-Engines.aspx